docs/AQL/grammar/AqlParser.g4

```diff
@@ -0,0 +1,214 @@
+//
```

Comment:
Yes you are right, we should try to reuse existing things.
For now I copied most of Archie's BaseLexer / base_patterns lexer rules with some modifications (and also removing lines AQL does not need), but we'll need to optimize it and properly use 'import'.

Reply:
It could be possible to re-use some simple lexer rules that really apply to all grammars. But reuse in ANTLR can cause many problems, and often it is just better to define lexers specific to that specific grammar. If even one lexer rule is not to be used in that particular grammar, it is probably already better to define a new lexer that does not contain that one rule.